language/move-vm/runtime/src/loader.rs

```diff
@@ -1442,6 +1378,193 @@ impl Loader {
         })
     }
 
+    fn struct_gidx_to_type_tag(&self, gidx: usize, ty_args: &[Type]) -> VMResult<StructTag> {
```

Comment:
I wonder if all this code and generally the code about the type cache could be moved elsewhere (different file) and leave the loader a bit more about loading.
We don't want it in this PR and it's more of a question if w could do it

Reply:
I kinda get what you mean here. I think this also depends on how we view the `Type` similar to the case for `TransactionDataCache`. Definitely something for us to think about.